Informatics II/001/Task 2/Yuirike Solution/Task2-GeneralCase.c

```diff
@@ -0,0 +1,58 @@
+#include <stdio.h>
+/*
+Test this code for general n x n cases
+If you use VSCode, you can run this with Code-Runner.
+*/
+
+//Function to calculate the dot product
+int dotP(int A[], int B[], int n)
+{
+    int res = 0;
+    for (int i = 0; i < n; i++)
+    {
+        for (int j = 0; j < n; j++)
+        {
+            if (i == j)
+            {
+                res = res + B[j] * A[i];
+            }
+        }
+    }
```

Comment:
This loop can be easily refactored into a single loop:
```suggestion
   for (int i = 0; i < n; i++)
    {
                res = res + B[i] * A[i];
    }
```

Reply:
Agree with @Patertuck (plus unindent line 13)

Reply:
> Agree with @Patertuck (plus unindent line 13)

oh yes, done

Reply:
I feel kind of stupid for not realizing that...